src/libraries/Microsoft.Extensions.Logging/src/LoggerInformation.cs

```diff
@@ -44,23 +44,23 @@ public bool IsEnabled(LogLevel level)
 
     internal readonly struct ScopeLogger
     {
-        public ScopeLogger(ILogger logger, IExternalScopeProvider externalScopeProvider)
+        public ScopeLogger(ILogger? logger, IExternalScopeProvider? externalScopeProvider)
         {
             Logger = logger;
             ExternalScopeProvider = externalScopeProvider;
         }
 
-        public ILogger Logger { get; }
+        public ILogger? Logger { get; }
 
-        public IExternalScopeProvider ExternalScopeProvider { get; }
+        public IExternalScopeProvider? ExternalScopeProvider { get; }
 
         public IDisposable CreateScope<TState>(TState state)
         {
             if (ExternalScopeProvider != null)
             {
                 return ExternalScopeProvider.Push(state);
             }
-            return Logger.BeginScope<TState>(state);
+            return Logger!.BeginScope<TState>(state);
```

Comment:
As you already pointed out in the PR description, there's already usage sample instantiating ScopeLogger with a null logger. So `Logger` here could be null.

https://github.com/dotnet/runtime/blob/78a2d2e0c49be1e1ae68e851b6df196304b6cee7/src/libraries/Microsoft.Extensions.Logging/src/LoggerFactory.cs#L235

Reply:
Nullable PRs are not generally allowed to change the behavior. So what should be done here? Just leave it as `return Logger!.BeginScope<TState>(state);`?

Also it cannot be `return Logger?.BeginScope<TState>(state);` since `CreateScope` cannot return null by contract (it's possible that `Logging.Anstrations` is annotated incorrectly though)

Reply:
@maryamariyan - thoughts here?

Reply:
>  (it's possible that Logging.Abstractions is annotated incorrectly though)

which contract are you referring to? You mean `IDisposable ILogger.BeginScope`?

Could we add a test that exercises the code path where we pass a null logger to ScopeLogger here? 

https://github.com/dotnet/runtime/blob/78a2d2e0c49be1e1ae68e851b6df196304b6cee7/src/libraries/Microsoft.Extensions.Logging/src/LoggerFactory.cs#L235

Reply:
> which contract are you referring to? You mean `IDisposable ILogger.BeginScope`?

Yes. On the other hand `loggers[0].CreateScope(state);` can be turned into `loggers[0].CreateScope(state) ?? NullScope.Instance;`

Reply:
Added the assert, but this doesn't look good.. Maybe at least constructor check should be added to ensure that at least one of the arguments is not null? (possibly in different PR)

Reply:
Here and the ctor sounds good to me.